internal/msalbase/authentication_result.go

```diff
@@ -5,82 +5,67 @@ package msalbase
+// CreateAuthenticationResultFromStorageTokenResponse creates an authenication result from a storage token response (which is generated from the cache).
+func CreateAuthenticationResultFromStorageTokenResponse(storageTokenResponse StorageTokenResponse) (AuthenticationResult, error) {
+	if storageTokenResponse.AccessToken == nil {
+		return AuthenticationResult{}, errors.New("no access token present in cache")
 	}
+
 	account := storageTokenResponse.account
-	var idToken *IDToken
-	accessToken := ""
-	var expiresOn time.Time
-	grantedScopes := []string{}
-	declinedScopes := []string{}
-	var err error
-	// Checking if the cache had an access token, if not, need to throw an error
-	if !reflect.ValueOf(storageTokenResponse.accessToken).IsNil() {
-		accessToken = storageTokenResponse.accessToken.GetSecret()
-		expiresOn, err = ConvertStrUnixToUTCTime(storageTokenResponse.accessToken.GetExpiresOn())
-		if err != nil {
-			return nil, errors.New("access token in cache expires at an invalid time")
-		}
-		grantedScopes = SplitScopes(storageTokenResponse.accessToken.GetScopes())
-	} else {
-		return nil, errors.New("no access token present in cache")
+	accessToken := storageTokenResponse.AccessToken.GetSecret()
+	expiresOn, err := ConvertStrUnixToUTCTime(storageTokenResponse.AccessToken.GetExpiresOn())
+	if err != nil {
+		return AuthenticationResult{}, errors.New("access token in cache expires at an invalid time")
 	}
-	// Checking if there was an ID token in the cache; this will throw an error in the case of confidential client applications
-	if !reflect.ValueOf(storageTokenResponse.idToken).IsNil() {
-		idToken, err = CreateIDToken(storageTokenResponse.idToken.GetSecret())
+	grantedScopes := SplitScopes(storageTokenResponse.AccessToken.GetScopes())
+
+	// Checking if there was an ID token in the cache; this will throw an error in the case of confidential client applications.
+	var idToken IDToken
+	if storageTokenResponse.IDToken != nil {
+		idToken, err = CreateIDToken(storageTokenResponse.IDToken.GetSecret())
 		if err != nil {
-			return nil, err
+			return AuthenticationResult{}, err
 		}
 	}
-	ar := &AuthenticationResult{account, idToken, accessToken, expiresOn, grantedScopes, declinedScopes}
-	return ar, nil
+	return AuthenticationResult{account, idToken, accessToken, expiresOn, grantedScopes, nil}, nil
 }
 
-// CreateAuthenticationResult creates an AuthenticationResult
-func CreateAuthenticationResult(tokenResponse *TokenResponse, account *Account) (*AuthenticationResult, error) {
-	grantedScopes := tokenResponse.GrantedScopes
-	declinedScopes := tokenResponse.declinedScopes
-	//Checking if there are any scopes that were declined, this would lead to an invalid access token
-	if len(declinedScopes) > 0 {
-		return nil, errors.New("token response failed because declined scopes are present")
+// CreateAuthenticationResult creates an AuthenticationResult.
+// TODO(jdoak): Make this a method on TokenResponse() that takes only 1 arge, Account.
```

Comment:
Please do the todo or delete it.

Reply:
I'm going to hold off on this for a big.  Many of these todos are notes for future edits.  This todo will happen in a future refactor.